### PR TITLE
Update to Django 1.10

### PR DESCRIPTION
**Pull request by rowanseymour**
Was running into conflicting dependencies so... have done an update of all dependencies and started using [pip-tools](https://github.com/nvie/pip-tools) to find conflicts and generate an annotated pip-freeze.txt that will be easier to update in future. We'll obviously test this pretty thoroughly before merging.

### CODE REVIEW

File: pip-requires.txt [hunk 1]
```diff
@@ -1,12 +1,11 @@
 boto
-celery[redis]
+celery-with-redis
 colorama
-confmodel
+-e git+https://github.com/nyaruka/confmodel.git@python3#egg=confmodel
```
Comment (rowanseymour) [inline pip-requires.txt]: Would be great to get a released version of this @smn @rudigiesler

File: casepro/utils/tests.py [hunk 2]
```diff
@@ -86,7 +86,7 @@ def to_json(self):
             MyClass()
         ]
 
-        self.assertEqual(json_encode(data), '["string", "2015-10-09T14:48:30.123456", "bar", {"bar": "X"}]')
+        self.assertEqual(json_encode(data), '["string", "2015-10-09T14:48:30.123456Z", "bar", {"bar": "X"}]')
```
Comment (rowanseymour) [inline casepro/utils/tests.py]: Due to updating to latest rapidpro client which provides the date formatting function

File: pip-requires.txt (hunk 1 above)
Comment (smn) [inline pip-requires.txt]: I hear you. Following up with my team on this.

Comment (rowanseymour) [inline pip-requires.txt]: Thanks!

**Pull request merged**